cumulusci/tasks/github/publish.py

```diff
@@ -0,0 +1,165 @@
+        super()._init_options(kwargs)
+        self.options["include"] = process_list_arg(
+            self.options.get(
+                "include", ["datasets/", "documentation/", "tasks/", "unpackaged/"]
+            )
+        )
+        if self.options["version"] == "latest":
+            self.options["version"] = str(self.project_config.get_latest_version())
+        elif self.options["version"] == "latest_beta":
+            self.options["version"] = str(
+                self.project_config.get_latest_version(beta=True)
+            )
+        self.options["create_release"] = process_bool_arg(
+            self.options.get("create_release", True)
+        )
+        self.options["release_body"] = process_bool_arg(
+            self.options.get("release_body", False)
+        )
+        self.options["dry_run"] = process_bool_arg(self.options.get("dry_run", False))
+
+    def _get_target_repo_api(self):
+        gh = self.project_config.get_github_api(
+            self._target_repo_info["owner"], self._target_repo_info["name"]
+        )
+        return gh.repository(
+            self._target_repo_info["owner"], self._target_repo_info["name"]
+        )
+
+    def _run_task(self):
+        self._target_repo_info = self.project_config._split_repo_url(
+            self.options["repo_url"]
+        )
+        self.target_repo = self._get_target_repo_api()
+
+        with tempfile.TemporaryDirectory() as target:
+            self._download_repo_and_extract(target)
+            commit = self._create_commit(target)
+            if commit and self.options["create_release"]:
+                self._create_release(target, commit.sha)
+
+    def _download_repo_and_extract(self, path):
+        zf = download_extract_github_from_repo(
+            self.get_repo(),
+            ref="tags/{}".format(
+                self.project_config.get_tag_for_version(self.options["version"])
+            ),
+        )
+        included_members = self._filter_namelist(
+            includes=self.options["include"], namelist=zf.namelist()
+        )
+        zf.extractall(path=path, members=included_members)
+
+    def _filter_namelist(self, includes, namelist):
+        filtered_names = set()
+        dirs = tuple(name for name in includes if name.endswith("/"))
+        filtered_names.update(
+            {name for name in namelist if name.startswith(dirs) or name in includes}
+        )
+        return list(filtered_names)
+
+    def _create_commit(self, path):
+        committer = CommitDir(self.target_repo, logger=self.logger)
+        message = f"Publishing release {self.options['version']}"
+        return committer(
+            path,
+            self.options["branch"],
+            repo_dir="",
+            commit_message=message,
+            dry_run=self.options["dry_run"],
+        )
+
+    def _create_release(self, path, commit):
+        # Get current release info
+        tag_name = self.project_config.get_tag_for_version(self.options["version"])
```

Comment:
This is happening in multiple places; you could set self.tag_name at the start of _run_task